### ISSUE TITLE
Requests pip package version in conflict

### ISSUE THREAD

**Issue by sophiasherman-capitalone** [thelastpickle/cassandra-medusa#710]
[Project board link](https://github.com/orgs/k8ssandra/projects/8/views/1?pane=issue&itemId=52715163)

These lines seem to be in direct conflict with each other.
https://github.com/thelastpickle/cassandra-medusa/blob/master/setup.py#L59
https://github.com/thelastpickle/cassandra-medusa/blob/master/requirements.txt#L16

Medusa version: 0.17.2
Python version: 3.11.5

requirements.txt file:
pip==23.1
boto==2.49.0
boto3==1.26.0
jmespath==0.10.0
ansible==9.1.0
passlib==1.7.4
tenacity==6.3.1
hvac==1.2.1
cassandra-medusa==0.17.2

Command: pip install -r requirements.txt:

Error:
```
ERROR: Could not find a version that satisfies the requirement pip==23.1 (from -r requirements.txt (line 1)) (from versions: 0.2, 0.2.1, 0.3, 0.3.1, 0.4, 0.5, 0.5.1, 0.6, 0.6.1, 0.6.2, 0.6.3, 0.7, 0.7.1, 0.7.2, 0.8, 0.8.1, 0.8.2, 0.8.3, 1.0, 1.0.1, 1.0.2, 1.1, 1.2, 1.2.1, 1.3, 1.3.1, 1.4, 1.4.1, 1.5, 1.5.1, 1.5.2, 1.5.3, 1.5.4, 1.5.5, 1.5.6, 6.0, 6.0.1, 6.0.2, 6.0.3, 6.0.4, 6.0.5, 6.0.6, 6.0.7, 6.0.8, 6.1.0, 6.1.1, 7.0.0, 7.0.1, 7.0.2, 7.0.3, 7.1.0, 7.1.1, 7.1.2, 8.0.0, 8.0.1, 8.0.2, 8.0.3, 8.1.0, 8.1.1, 8.1.2, 9.0.0, 9.0.1, 9.0.2, 9.0.3, 10.0.0b1, 10.0.0b2, 10.0.0, 10.0.1, 18.0, 18.1, 19.0, 19.0.1, 19.0.2, 19.0.3, 19.1, 19.1.1, 19.2, 19.2.1, 19.2.2, 19.2.3, 19.3, 19.3.1, 20.0, 20.0.1, 20.0.2, 20.1b1, 20.1, 20.1.1, 20.2b1, 20.2, 20.2.1, 20.2.2, 20.2.3, 20.2.4, 20.3b1, 20.3, 20.3.1, 20.3.2, 20.3.3, 20.3.4)
ERROR: No matching distribution found for pip==23.1 (from -r requirements.txt (line 1))
```

Command: pip3 install -r requirements.txt:

Error:
```
ERROR: Cannot install -r requirements.txt (line 8) and -r requirements.txt (line 9) because these package versions have conflicting dependencies.

The conflict is caused by:
    hvac 1.2.1 depends on requests<3.0.0 and >=2.27.1
    cassandra-medusa 0.17.2 depends on requests==2.22.0
```

Potential fix: Updating this line https://github.com/thelastpickle/cassandra-medusa/blob/master/setup.py#L59 to 'requests>=2.22.0'

**Comment by criscodex** [comment]
@sophiasherman-capitalone this will likely be fixed by #683

**Comment by cmlicata** [comment]
@criscodex, when do we expect #683  to be merged and released?
